### DISCUSSION TITLE
problems installing old version of cmake

### DISCUSSION THREAD

**Discussion by freakalot** [Homebrew/discussions#2102] [Everyday usage] [answered]
Hi guys, 

I try to install a previous version of cmake (3.18.4). We do have a need to keep the versions installed on our build machines stable and not just blindly install the latest version. I extracted the formula into my own tap to save it with

`brew extract --version 3.18.4 cmake homebrew/mytap`

using the following command to install it:

`brew install homebrew/mytap/cmake@3.18.4`

The build process ended up with the folowwing error:

`==> Downloading https://github.com/Kitware/CMake/releases/download/v3.18.4/cmake-3.18.4.tar.gz
Already downloaded: /Users/beep/Library/Caches/Homebrew/downloads/bb11533563004cdfcffb2210c073677f4146dbea89c1b48e9471a96cb1be0069--cmake-3.18.4.tar.gz
==> Installing cmake@3.18.4 from homebrew/lsyzrh
==> ./bootstrap --prefix=/usr/local/Cellar/cmake@3.18.4/3.18.4 --no-system-libs --parallel=12 --datadir=/share/cmake --docdir=/share/doc/cmake --mandir=/share/man --sphinx-build=/usr/local/opt/sphinx-doc/bin/sphinx-build --sphinx-html --sphinx-man --system-zlib --system-bzip2 --system-curl -- -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/cmake@3.18.4/3.18.4 -DCMAKE
==> make
Last 15 lines from /Users/beep/Library/Logs/Homebrew/cmake@3.18.4/02.make:
/usr/bin/ranlib libcmsys.a
[ 30%] Built target cmsys
[ 31%] Linking C static library libcmliblzma.a
cd /tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Utilities/cmliblzma && /private/tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Bootstrap.cmk/cmake -P CMakeFiles/cmliblzma.dir/cmake_clean_target.cmake
cd /tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Utilities/cmliblzma && /private/tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Bootstrap.cmk/cmake -E cmake_link_script CMakeFiles/cmliblzma.dir/link.txt --verbose=1
/usr/bin/ar qc libcmliblzma.a CMakeFiles/cmliblzma.dir/liblzma/check/check.c.o CMakeFiles/cmliblzma.dir/liblzma/check/crc32_fast.c.o CMakeFiles/cmliblzma.dir/liblzma/check/crc32_table.c.o CMakeFiles/cmliblzma.dir/liblzma/check/crc64_fast.c.o CMakeFiles/cmliblzma.dir/liblzma/check/crc64_table.c.o CMakeFiles/cmliblzma.dir/liblzma/check/sha256.c.o CMakeFiles/cmliblzma.dir/liblzma/common/alone_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/alone_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/auto_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/block_buffer_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/block_buffer_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/block_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/block_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/block_header_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/block_header_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/block_util.c.o CMakeFiles/cmliblzma.dir/liblzma/common/common.c.o CMakeFiles/cmliblzma.dir/liblzma/common/easy_buffer_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/easy_decoder_memusage.c.o CMakeFiles/cmliblzma.dir/liblzma/common/easy_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/easy_encoder_memusage.c.o CMakeFiles/cmliblzma.dir/liblzma/common/easy_preset.c.o CMakeFiles/cmliblzma.dir/liblzma/common/filter_buffer_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/filter_buffer_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/filter_common.c.o CMakeFiles/cmliblzma.dir/liblzma/common/filter_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/filter_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/filter_flags_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/filter_flags_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/index.c.o CMakeFiles/cmliblzma.dir/liblzma/common/index_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/index_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/index_hash.c.o CMakeFiles/cmliblzma.dir/liblzma/common/stream_buffer_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/stream_buffer_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/stream_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/stream_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/stream_flags_common.c.o CMakeFiles/cmliblzma.dir/liblzma/common/stream_flags_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/stream_flags_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/vli_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/vli_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/common/vli_size.c.o CMakeFiles/cmliblzma.dir/liblzma/delta/delta_common.c.o CMakeFiles/cmliblzma.dir/liblzma/delta/delta_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/delta/delta_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/lz/lz_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/lz/lz_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/lz/lz_encoder_mf.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/fastpos_table.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/lzma2_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/lzma2_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/lzma_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/lzma_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/lzma_encoder_optimum_fast.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/lzma_encoder_optimum_normal.c.o CMakeFiles/cmliblzma.dir/liblzma/lzma/lzma_encoder_presets.c.o CMakeFiles/cmliblzma.dir/liblzma/rangecoder/price_table.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/arm.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/armthumb.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/ia64.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/powerpc.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/simple_coder.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/simple_decoder.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/simple_encoder.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/sparc.c.o CMakeFiles/cmliblzma.dir/liblzma/simple/x86.c.o
/usr/bin/ranlib libcmliblzma.a
[ 31%] Built target cmliblzma
[ 31%] Linking C static library libcmzstd.a
cd /tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Utilities/cmzstd && /private/tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Bootstrap.cmk/cmake -P CMakeFiles/cmzstd.dir/cmake_clean_target.cmake
cd /tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Utilities/cmzstd && /private/tmp/cmakeA3.18.4-20210907-24179-btudg2/cmake-3.18.4/Bootstrap.cmk/cmake -E cmake_link_script CMakeFiles/cmzstd.dir/link.txt --verbose=1
/usr/bin/ar qc libcmzstd.a CMakeFiles/cmzstd.dir/lib/common/entropy_common.c.o CMakeFiles/cmzstd.dir/lib/common/error_private.c.o CMakeFiles/cmzstd.dir/lib/common/fse_decompress.c.o CMakeFiles/cmzstd.dir/lib/common/pool.c.o CMakeFiles/cmzstd.dir/lib/common/threading.c.o CMakeFiles/cmzstd.dir/lib/common/xxhash.c.o CMakeFiles/cmzstd.dir/lib/common/zstd_common.c.o CMakeFiles/cmzstd.dir/lib/compress/fse_compress.c.o CMakeFiles/cmzstd.dir/lib/compress/hist.c.o CMakeFiles/cmzstd.dir/lib/compress/huf_compress.c.o CMakeFiles/cmzstd.dir/lib/compress/zstd_compress.c.o CMakeFiles/cmzstd.dir/lib/compress/zstd_double_fast.c.o CMakeFiles/cmzstd.dir/lib/compress/zstd_fast.c.o CMakeFiles/cmzstd.dir/lib/compress/zstd_lazy.c.o CMakeFiles/cmzstd.dir/lib/compress/zstd_ldm.c.o CMakeFiles/cmzstd.dir/lib/compress/zstdmt_compress.c.o CMakeFiles/cmzstd.dir/lib/compress/zstd_opt.c.o CMakeFiles/cmzstd.dir/lib/decompress/huf_decompress.c.o CMakeFiles/cmzstd.dir/lib/decompress/zstd_ddict.c.o CMakeFiles/cmzstd.dir/lib/decompress/zstd_decompress_block.c.o CMakeFiles/cmzstd.dir/lib/decompress/zstd_decompress.c.o CMakeFiles/cmzstd.dir/lib/deprecated/zbuff_common.c.o CMakeFiles/cmzstd.dir/lib/deprecated/zbuff_compress.c.o CMakeFiles/cmzstd.dir/lib/deprecated/zbuff_decompress.c.o CMakeFiles/cmzstd.dir/lib/dictBuilder/cover.c.o CMakeFiles/cmzstd.dir/lib/dictBuilder/divsufsort.c.o CMakeFiles/cmzstd.dir/lib/dictBuilder/fastcover.c.o CMakeFiles/cmzstd.dir/lib/dictBuilder/zdict.c.o
/usr/bin/ranlib libcmzstd.a
[ 31%] Built target cmzstd
make: *** [all] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

Traceback (most recent call last):
	26: from /usr/local/Homebrew/Library/Homebrew/build.rb:229:in `<main>'
	25: from /usr/local/Homebrew/Library/Homebrew/build.rb:133:in `install'
	24: from /usr/local/Homebrew/Library/Homebrew/utils.rb:550:in `with_env'
	23: from /usr/local/Homebrew/Library/Homebrew/build.rb:138:in `block in install'
	22: from /usr/local/Homebrew/Library/Homebrew/formula.rb:1275:in `brew'
	21: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2434:in `stage'
	20: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/forwardable.rb:230:in `stage'
	19: from /usr/local/Homebrew/Library/Homebrew/resource.rb:91:in `stage'
	18: from /usr/local/Homebrew/Library/Homebrew/resource.rb:116:in `unpack'
	17: from /usr/local/Homebrew/Library/Homebrew/resource.rb:199:in `mktemp'
	16: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `run'
	15: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `chdir'
	14: from /usr/local/Homebrew/Library/Homebrew/mktemp.rb:63:in `block in run'
	13: from /usr/local/Homebrew/Library/Homebrew/resource.rb:117:in `block in unpack'
	12: from /usr/local/Homebrew/Library/Homebrew/download_strategy.rb:102:in `stage'
	11: from /usr/local/Homebrew/Library/Homebrew/download_strategy.rb:115:in `chdir'
	10: from /usr/local/Homebrew/Library/Homebrew/download_strategy.rb:115:in `chdir'
	 9: from /usr/local/Homebrew/Library/Homebrew/resource.rb:121:in `block (2 levels) in unpack'
	 8: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2454:in `block in stage'
	 7: from /usr/local/Homebrew/Library/Homebrew/utils.rb:550:in `with_env'
	 6: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2455:in `block (2 levels) in stage'
	 5: from /usr/local/Homebrew/Library/Homebrew/formula.rb:1282:in `block in brew'
	 4: from /usr/local/Homebrew/Library/Homebrew/build.rb:178:in `block (2 levels) in install'
	 3: from /usr/local/Homebrew/Library/Taps/homebrew/homebrew-lsyzrh/Formula/cmake@3.18.4.rb:44:in `install'
	 2: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2225:in `system'
	 1: from /usr/local/Homebrew/Library/Homebrew/formula.rb:2225:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:2289:in `block in system': Failed executing: make (BuildError)
	9: from /usr/local/Homebrew/Library/Homebrew/brew.rb:162:in `<main>'
	8: from /usr/local/Homebrew/Library/Homebrew/brew.rb:174:in `rescue in <main>'
	7: from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:509:in `dump'
	6: from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:455:in `issues'
	5: from /usr/local/Homebrew/Library/Homebrew/exceptions.rb:459:in `fetch_issues'
	4: from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:61:in `issues_for_formula'
	3: from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:30:in `search_issues'
	2: from /usr/local/Homebrew/Library/Homebrew/utils/github.rb:160:in `search'
	1: from /usr/local/Homebrew/Library/Homebrew/utils/github/api.rb:234:in `open_rest'
/usr/local/Homebrew/Library/Homebrew/utils/github/api.rb:304:in `raise_error': Validation Failed: [{"message"=>"The listed users and repositories cannot be searched either because the resources do not exist or you do not have permission to view them.", "resource"=>"Search", "field"=>"q", "code"=>"invalid"}] (GitHub::API::ValidationFailedError)
`

I'm really not sure what it tries to check on github here and which repository is affected. When I run make not via brew, it is able to build this target.

**Comment by carlocab** [comment, marked as answer]
`brew` tries to check the relevant GitHub repository for related issues or pull requests in the event of build failure. It didn't manage to do that, so it errored out. However, fixing that error won't really help you do what you want, since what you want to find is the cause of the build failure. You'll need to check the logs in the path shown to you (`/Users/beep/Library/Logs/Homebrew/cmake@3.18.4/02.make`) to see what happened.

**Reply by freakalot** [reply]
ok, I found the issue. the dependency pointed to a specific version was called in the cmake formula as a parameter for the bootstrap. This also needed to be adjusted.

Good to know why it tries to access github in case of an error and how to understand this error.
